vite.config.js

```diff
@@ -0,0 +1,19 @@
+import { resolve } from 'path'
+import { defineConfig } from 'vite'
+import { nodeResolve } from '@rollup/plugin-node-resolve'
+
+export default defineConfig({
+  plugins: [nodeResolve()],
```

Comment:
From https://github.com/melange-re/melange-for-react-devs-template/pull/1/files/afb748d3a47351917943ed43fd0376901a111aff#r1450604037

> I asked in the other PR, but why is nodeResolve needed?

Without it, Vite is unable to import the modules located in, say, `_build/default/src/counter/output/node_modules`. The error you would get is:

```
10:19:45 AM [vite] Internal server error: Failed to resolve import "melange.js/caml_option.js" from "_build/default/src/counter/output/src/counter/Index.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/fhsu/work/melange-for-react-devs/_build/default/src/counter/output/src/counter/Index.js:3:31
  1  |  // Generated by Melange
  2  |  
  3  |  import * as Caml_option from "melange.js/caml_option.js";
     |                                ^
  4  |  import * as Melange__Counter from "./Counter.js";
  5  |  import * as ReactDom from "react-dom";
      at formatError (file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:63765:46)
      at TransformContext.error (file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:63759:19)
      at normalizeUrl (file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:62034:33)
      at async file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:62188:47
      at async Promise.all (index 0)
      at async TransformContext.transform (file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:62109:13)
      at async Object.transform (file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:64060:30)
      at async loadAndTransform (file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:49741:29)
      at async viteTransformMiddleware (file:///home/fhsu/work/melange-for-react-devs/node_modules/vite/dist/node/chunks/dep-V3BH7oO1.js:59342:32)
10:19:45 AM [vite] Pre-transform error: Failed to resolve import "melange.js/curry.js" from "_build/default/src/counter/output/src/counter/Counter.js". Does the file exist?
```
I guess the default behavior of rollup is to ignore everything in `node_modules`, and this plugin allows you override that

Reply:
@jchavarri Should I add a comment here?

Reply:
Yeah, I think a comment would be helpful. I double checked with @davesnx in case there were other approaches to solve this issue, but apparently this is the way to go (even if it's weird for me to use a Node plugin for an app that is built for the browser...)

Reply:
Actually, maybe we should use `browser:true` here? https://github.com/rollup/plugins/tree/master/packages/node-resolve/#browser

Reply:
@jchavarri Sorry, I read the docs for `browser` option and I still don't quite understand what it does :disappointed:

Reply:
Added comment in https://github.com/melange-re/melange-for-react-devs/pull/16/commits/bd09d27b7f159197740485600d186e02df83fd01

Reply:
It refers to [this field](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#browser) in `package.json` which should be used by JS tooling to resolve packages that are being bundled for the browser (when a given package has different implementations for Node and browser).